ns3_adapter/src/ns3_client.cpp

```diff
@@ -13,86 +14,134 @@ NS3Client::~NS3Client() {
     }catch(...){}
 }
 
-bool NS3Client::registermsg(const std::shared_ptr<std::vector<uint8_t>>&message, const std::string &remote_address, unsigned short remote_port,
-                            unsigned short local_port)
-{
-    bool success = false;
-    
-    if (connect(remote_address, remote_port, local_port))
+
+bool NS3Client::connect_registration(const std::string &remote_address, unsigned short remote_port) {
+    boost::system::error_code ignored_ec;
+     //If we are already connected return false
+    if(running_) return false;
+    //Get remote endpoint
+    try
     {
-        bool send_success = sendNS3Message(message);
-        if (send_success)
-        { 
-            // ROS_DEBUG_STREAM("Handshake Message sent successfully");
-            success = true;
-        }
-        // else ROS_DEBUG_STREAM("Handshake Message send failed");
+        remote_udp_ep_registration_ = boost::asio::ip::udp::endpoint(boost::asio::ip::address::from_string(remote_address),remote_port);
+    }catch(std::exception e)
+    {
+        ignored_ec = boost::asio::error::invalid_argument;
+        throw e;
     }
-    // else ROS_DEBUG_STREAM( "Connection failed" );
-    else std::cerr << "Connection failed" << std::endl;
 
+    ignored_ec.clear();
 
-    // close();
-    
-    return success;
-}
-
+    io_.reset(new boost::asio::io_service());
+    output_strand_.reset(new boost::asio::io_service::strand(*io_));
+    udp_out_registration_socket_.reset(new boost::asio::ip::udp::socket(*io_,remote_udp_ep_registration_.protocol()));
+    udp_out_broadcasting_socket_.reset(new boost::asio::ip::udp::socket(*io_,remote_udp_ep_broadcasting_.protocol()));
+    work_.reset(new boost::asio::io_service::work(*io_));
 
-bool NS3Client::connect(const std::string &remote_address, unsigned short remote_port,
-                            unsigned short local_port) {
-    boost::system::error_code ignored_ec;
-    return connect(remote_address, remote_port, local_port, ignored_ec);
+    io_thread_.reset(new std::thread([this]()
+                                     {
+                                         boost::system::error_code err;
+                                         io_->run(err);
+                                         if(err)
+                                         {
+                                             onError(err);
+                                         }
+                                     }));
+    running_ = true;
+    onConnect();
+    return true;
```

Comment:
Can we not just return running?

Reply:
the client is setting the running_ = true just here, so they are identical